client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.html

```diff
@@ -0,0 +1,111 @@
+        uib-tooltip="{{ 'Note: You have to reject all change recommendations if the plenum does not follow the recommendation. This does not affect amendments.' | translate }}"
+        ng-click="viewChangeRecommendations.rejectAllChangeRecommendations(motion)">
+      <i class="fa fa-thumbs-down"></i>
+      <translate>Reject all change recommendations</translate>
+    </button>
+    -->
+
+    <ul *ngIf="changes.length > 0">
+        <li *ngFor="let change of changes">
+          <a href='' (click)="scrollToChangeClicked(change, $event)"
+             [class.amendment]="isAmendment(change)"
+             [class.recommendation]="isChangeRecommendation(change)">
+            <span *ngIf="change.getLineFrom() >= change.getLineTo() - 1" class="line-number">
+                Line {{ change.getLineFrom() }}<span *ngIf="isChangeRecommendation(change)"></span>
+            </span>
+            <span *ngIf="change.getLineFrom() < change.getLineTo() - 1" class="line-number">
+                Line {{ change.getLineFrom() }} -
+                    {{ change.getLineTo() - 1 }}<span *ngIf="isChangeRecommendation(change)"></span>
+            </span>
+            <span *ngIf="isChangeRecommendation(change)">(Change recommendation)</span>
+            <span *ngIf="isAmendment(change)">@TODO Identifier</span>
+            <span class="operation" *ngIf="isChangeRecommendation(change)">–
+                {{ getRecommendationTypeName(change) }}
+                <!-- @TODO
+                <span ng-if="change.original.getType(motion.getVersion(version).text) == 3">
+                  { change.other_description }
+                </span>
+                -->
+            </span>
+            <span class="status">
+                <ng-container *ngIf="change.isRejected()">Rejected</ng-container>
+                <ng-container *ngIf="change.isAccepted() && isAmendment(change)">Accepted</ng-container>
+            </span>
+          </a>
+        </li>
+    </ul>
+
+    <div *ngIf="changes.length === 0" class="no-changes">
+        No change recommendations yet
+    </div>
+</section>
+
+
+<!-- The actual diff view -->
+<div class="motion-text-with-diffs">
+    <div *ngFor="let change of changes; let i = index">
+        <div class="motion-text original-text line-numbers-outside"
+             *ngIf="i === 0 || changes[i - 1].getLineTo() < change.getLineFrom()"
+             [innerHTML]="getTextBetweenChanges(changes[i - 1], change)"></div>
+
+        <div [class.collides]="hasCollissions(change)"
+             class="diff-box diff-box-{{ change.id }} clearfix">
+            <div class="collission-hint" *ngIf="hasCollissions(change)">
+                <!-- @TODO Title should be a tooltip -->
+                <i class="fa fa-warning" title="This change collides with another one."></i>
+            </div>
+            <div class="action-row">
+                <ng-container *ngIf="admin">
```

Comment:
The link introduces ng-container like this:
```
The ng-container directive
In order to avoid having to create that extra div, we can instead use ng-container directive
```
Unless ng-container has disadvantages, I would rather ask, do we `have to` create a DIV? (Though, admittedly, my code here already smells like a DIV-soup, so one could argue that it doesn't matter anymore anyway... :) )

Reply:
We should use as much `ng-container` as possible to keep the amount of elements low. Only use another element, if e.g. styles should be attached. So: OK!